kafka-rest/src/main/java/io/confluent/kafkarest/controllers/PartitionManagerImpl.java

```diff
@@ -73,7 +79,23 @@ public CompletableFuture<Optional<Partition>> getPartition(
       String clusterId, String topicName, int partitionId) {
     return topicManager
         .getTopic(clusterId, topicName)
-        .thenApply(topic -> checkEntityExists(topic, "Topic %s cannot be found.", topic))
+        .handle(
+            (value, exception) -> {
+              if (exception != null) {
+                log.error("Caught exception while get topic from cluster", exception);
+                if (exception.getCause() instanceof UnknownTopicOrPartitionException) {
+                  String exceptionMessage =
+                      String.format(
+                          "This server does not host this topic-partition for topic %s", topicName);
+                  throw new UnknownTopicOrPartitionException(exceptionMessage, exception);
+                } else if (exception instanceof NotFoundException
```

Comment:
Why is this extra check for the NotFoundException needed?  Can we not just rethrow the original exception? This seems to be adding an extra layer of NotFoundException around the CompletionException.

Reply:
The type of exception here is `Throwable`, we can't re-throw it without handle it. There are some test cases related to `NotFoundException`, we need to take it out to handle them directly.

<img width="653" alt="Screen Shot 2022-09-08 at 10 45 53 PM" src="https://user-images.githubusercontent.com/7269552/189279916-3bb95ffd-b397-4c33-882b-f5f4abea74ae.png">